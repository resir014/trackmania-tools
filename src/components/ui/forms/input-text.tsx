import * as React from 'react';

import clsx from 'clsx';

interface InputTextProps extends React.InputHTMLAttributes<HTMLInputElement> {
  isGroupItem?: boolean;
}

export const InputText = React.forwardRef<HTMLInputElement, InputTextProps>(
  ({ className, type = 'text', isGroupItem, ...rest }, ref) => {
    return (
      <input
        className={clsx(
          isGroupItem ? 'first:rounded-l-md last:rounded-r-md' : 'shadow-sm rounded-md',
          'block w-full text-sm border-transparent bg-gray-800 focus:ring-green-500 focus:border-green-500',
          className
        )}
        ref={ref}
        type={type}
        {...rest}
      />
    );
  }
);

InputText.displayName = 'InputText';
