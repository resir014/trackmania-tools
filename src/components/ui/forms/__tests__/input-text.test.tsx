import * as React from 'react';
import { render } from '@testing-library/react';
import { InputText } from '../input-text';

describe('InputText', () => {
  it('renders correctly', () => {
    const { container } = render(<InputText />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
