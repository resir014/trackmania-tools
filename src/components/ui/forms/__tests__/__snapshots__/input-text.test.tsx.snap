// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputText renders correctly 1`] = `
<input
  class="shadow-sm rounded-md block w-full text-sm border-transparent bg-gray-800 focus:ring-green-500 focus:border-green-500"
  type="text"
/>
`;
